### ISSUE TITLE
Switch from V1 resolver API to v2 balancer API

### ISSUE THREAD

**Issue by jadekler** [googleapis/google-api-go-client#341]
The grpc v1 resolver/balancer API is deprecated: https://godoc.org/google.golang.org/grpc#WithBalancer.

We should switch to the v2 balancer API (no resolver needed): https://godoc.org/google.golang.org/grpc/balancer.

However, in order to pass along the num conns setting, we need local service config, which is coming soon (tm). So, this is blocked until then.

**Comment by jadekler** [comment]
Blocked on https://github.com/grpc/grpc-go/pull/2686 and a separate, future PR that lets the load balancer receive the service config.

**Comment by codyoss** [comment]
This is no longer relevant as of: https://code-review.googlesource.com/c/google-api-go-client/+/50573. Closing.
